akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -550,4 +550,19 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
     !serializedClass.getName.startsWith("java.lang.") &&
     !suppressWarningOnNonSerializationVerification(serializedClass)
   }
+
+  /**
+   * INTERNAL API: Test utility to verify serialization roundtrip.
+   * Exposed as public utility from TestKits.
+   * Throws exception from serializer if `obj` can't be serialized and
+   * deserialized.
+   *
+   * @return the deserialized object
+   */
+  @InternalApi private[akka] def verifySerialization(obj: AnyRef): AnyRef = {
+    val bytes = serialize(obj).get
+    val serializer = findSerializerFor(obj)
+    val manifest = Serializers.manifestFor(serializer, obj)
+    deserialize(bytes, serializer.identifier, manifest).get
+  }
```

Comment:
Since it is only for tests, why can't it live only in test kit?

Reply:
needed it in InMemJournal too, and we probably want it from both typed and untyped testkits.

Reply:
The typed testkit depends on the untyped so it can just delegate?

Reply:
I see no reason for having that coupling, created issue https://github.com/akka/akka/issues/27303

Reply:
Ok, In that case I'd still prefer duplicating the logic to having testkit stuff in the non-testkit codebase.

Reply:
ok